ingest/src/main/java/com/microsoft/azure/kusto/ingest/resources/RankedStorageAccount.java

```diff
@@ -42,7 +45,12 @@ public void addResult(boolean success) {
     private Bucket adjustForTimePassed() {
         if (buckets.isEmpty()) {
```

Comment:
ou don't need a lock here?

Reply:
We should if we're concerned about race conditions, which I don't think we are.

In terms of preventing `ConcurrentModificationException`, I don't think locking here would help, because `ArrayDeque.isEmpty()`'s implementation is simply `return head == tail;`.